src/test/java/ch/uzh/ifi/hase/soprafs22/service/UserServiceTest.java

```diff
@@ -165,9 +263,413 @@ void updateUser_Conflict(){ //Test what happens when user tries to change userna
+        long id = testUser.getId();
+
+        Mockito.when(userRepository.findById(id)).thenReturn(testUser);
+        assertFalse(userService.isWhiteCardBelongingToUser(testWhiteCard, testUser.getId()));
+    }
+
+    @Test
+    void deleteWhiteCard(){
+      //Create a second white card to delete
+        WhiteCard deleteCard = new WhiteCard();
+        deleteCard.setId(1L);
+        deleteCard.setText("CardText");
+        testWhiteCards.add(deleteCard); //Add the card to delete to the whitecards for the user
+        testUser.setUserWhiteCards(testWhiteCards);
+
+        long id = testUser.getId();
+        Mockito.when(userRepository.findById(id)).thenReturn(testUser);
+
+        userService.deleteWhiteCard(testUser.getId(), deleteCard);
+
+        assertTrue(userService.isWhiteCardBelongingToUser(testWhiteCard, testUser.getId()));//should still be there
+        assertFalse(userService.isWhiteCardBelongingToUser(deleteCard, testUser.getId())); //should be deleted now
+    }
+
+    @Test
+    void doesMatchExist_true(){
+        Match testMatch =  userService.createMatch(testUser, otherUser);
+        testMatch.setMatchId(2L);
+
+        Mockito.when(matchRepository.findByMatchId(2L)).thenReturn(testMatch);
+        //both users should now have the testmatch, so it should exist
+        userService.setMatch(testMatch);
+        assertTrue(userService.doesMatchExist(testUser, otherUser));
+        assertTrue(userService.doesMatchExist(otherUser, testUser));
+
+        //case multiple matches exist
+        User thirdUser = new User();
+        thirdUser.setId(13L);
+        Match otherMatch = userService.createMatch(testUser, thirdUser);
+        otherMatch.setMatchId(23L);
+
+        Mockito.when(matchRepository.findByMatchId(23L)).thenReturn(otherMatch);
+        userService.setMatch(otherMatch);
+        assertTrue(userService.doesMatchExist(testUser, thirdUser));
+        assertTrue(userService.doesMatchExist(thirdUser,testUser));
+    }
+
+    @Test
+    void doesMatchExist_false(){
+        // given
+        User testUser2 = new User();
+        testUser2.setId(3L);
+        testUser2.setName("testName");
+        testUser2.setUsername("testUsername3");
+        testUser2.setPassword("1234");
+        testUser2.setGender(Gender.OTHER);
+
+        Match testMatch = userService.createMatch(otherUser, testUser2);
+        testMatch.setMatchId(2L);
+
+        Mockito.when(matchRepository.findByMatchId(2L)).thenReturn(testMatch);
+        userService.setMatch(testMatch);
+        assertFalse(userService.doesMatchExist(testUser, otherUser));
+        assertFalse(userService.doesMatchExist(otherUser, testUser));
+    }
+
+    @Test
+    void checkGeneralAccess_granted(){
+      testUser.setToken("token");
+      Mockito.when(userRepository.findByToken(testUser.getToken())).thenReturn(testUser);
+      userService.checkGeneralAccess(testUser.getToken());
+      Mockito.verify(userRepository, Mockito.times(1)).findByToken(Mockito.any());
```

Comment:
Could you explain in a comment what verify does? or maybe link a resource or idk.

Reply:
verify verifies that there is a call in this case there is one call to userrepository.findbytoken(any parameter)

Reply:
Great, thanks 💯